### ISSUE TITLE
path.relative should strip trailing '/' for Node.js compat

### ISSUE THREAD

**Issue by strager** [oven-sh/bun#5440]
### What version of Bun is running?

1.0.1+31aec4ebe325982fc0ef27498984b0ad9969162b

### What platform is your computer?

Linux 5.19.0-46-generic x86_64 x86_64

### What steps can reproduce the bug?

```javascript
import path from 'node:path';
console.log(path.relative('foo/', 'foo/bar/'));
```

### What is the expected behavior?

```shell
$ node relative.mjs 
bar
```

### What do you see instead?

```shell
$ bun run relative.mjs 
bar/
```

**Comment by Hanaasagi** [comment]
I think this has been fixed in the latest version, thanks report.

**Comment by Jarred-Sumner** [comment]
This was fixed in Bun v1.0.19
